test/AWS.jl

```diff
@@ -944,23 +939,22 @@ end
 
         try
             # POST
-            tags = Dict("Tags" => LittleDict("Tag-01" => "Tag-01", "Tag-02" => "Tag-02"))
+            tags = Dict("Tag-01" => "Tag-01", "Tag-02" => "Tag-02")
 
             for vault in vault_names
-                Glacier.add_tags_to_vault("-", vault, tags)
+                Glacier.add_tags_to_vault("-", vault; Tags=tags)
```

Comment:
Appears this is set incorrectly according to the glacier mapping:
```suggestion
                Glacier.add_tags_to_vault("-", vault; tags=tags)
```

Reply:
If this isn't failing we should probably investigate why... Thinking about it more it's because `amazonify` just translates names and doesn't cause a failure if you use the actual names used in the `params` dict. I don't think we should support both and should fail if a key is provided that isn't in the mapping

Reply:
In the couple of lines above we were passing in,

> tags = Dict("Tags" => LittleDict("Tag-01" => "Tag-01", "Tag-02" => "Tag-02"))

This worked well because Glacier was expecting `Tags` at the top-level value in the structure. Now however since we're using snake_case kwargs we have a translation inside of `src/services/glacier.jl` that specifies,

> tags => Tags

So we need to pass in this structure as `tags=tags`, this will then convert the high-level object to `Tags` and continue along as it previously worked when making the request.

Reply:
@iamed2 This is another example of inconsistencies in the operation definitions. Every other Glacier optional parameter is specified in camel case EXCEPT for `Tags`.

```julia
"part_size" => "x-amz-part-size",
"job_parameters" => "jobParameters",
"archive_size" => "x-amz-archive-size",
"checksum" => "x-amz-sha256-tree-hash",
"range" => "Range",
"completed" => "completed",
"archive_description" => "x-amz-archive-description",
"body" => "body",
"policy" => "policy",
"vault_notification_config" => "vaultNotificationConfig",
"tag_keys" => "TagKeys",
"statuscode" => "statuscode",
"marker" => "marker",
"tags" => "Tags",
"limit" => "limit",
```

Reply:
Sure but in this case the difference is clearly communicated in the service spec, and there's nothing special necessary to handle this case as it follows the standard `rest-json` protocol.

Reply:
Correct, but in the current implementation in this package you as a user need to know that this specific parameter is the only one with a capitalization. If you make a request now you'll get some vague error back from AWS saying that the structure is malformed.

Going forward with these changes or like-wise ones, it's not a problem. But it's more-so to just highlight that there are inconsistencies all throughout the SDK, and it wouldn't surprise me if there's another service in a situation like EC2.

Reply:
> Correct, but in the current implementation in this package you as a user need to know that this specific parameter is the only one with a capitalization.

This is clear in the SDK docs though. I don't know how a user would expect to know what each argument is called without checking the docs. And it's consistent between boto3 and the PHP and JS SDKs, which both match how we pass arguments (as kwargs or a dict). The SDKs which are different (Go, C++, Ruby, Java) standardize casing across every type of service. The Ruby SDK is notably the only official SDK which passes a generic map of arguments AND standardizes casing.

Reply:
I don't think it's a stretch to assume that if the parameter is `tags` for one service it would be that for another and not `Tags`.

My issue here is that all of these inconsistencies don't inspire confidence that this package will just work out of the box for every service.

Reply:
> My issue here is that all of these inconsistencies don't inspire confidence that this package will just work out of the box for every service.

But as we progress to implementing the specs more correctly, we make progress. So far there has been exactly one case (EC2) where the spec did not appear to match required behaviour. This glacier situation was an example of the package working perfectly. This PR changed what the API expected, and so we needed to change what the tests passed.